pkg/util/secret_test.go

```diff
@@ -0,0 +1,153 @@
+			Namespace: "ns",
+			Name:      "pull-secret",
+			Labels:    map[string]string{"dptp.openshift.io/requester": "foo"},
+		},
+		Data: map[string][]byte{corev1.DockerConfigJsonKey: []byte("xyz")},
+		Type: corev1.SecretTypeDockerConfigJson,
+	}
+
+	srcSecretWithDiffLabel := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "pull-secret",
+			Labels:    map[string]string{"dptp.openshift.io/requester": "foo"},
+		},
+		Data: map[string][]byte{corev1.DockerConfigJsonKey: []byte("abc")},
+		Type: corev1.SecretTypeDockerConfigJson,
+	}
+
+	dstSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "ns",
+			Name:      "pull-secret",
+			Labels:    map[string]string{"dptp.openshift.io/requester": "bar"},
+		},
+		Data: map[string][]byte{corev1.DockerConfigJsonKey: []byte("abc")},
+		Type: corev1.SecretTypeDockerConfigJson,
+	}
+
+	im := true
+
+	testCases := []struct {
+		name          string
+		client        ctrlruntimeclient.Client
+		expected      bool
+		expectedError error
+		verify        func(ctrlruntimeclient.Client) error
+	}{
+		{
+			name:     "the target secret is created",
+			client:   fakeclient.NewFakeClient(),
+			expected: true,
+			verify: func(client ctrlruntimeclient.Client) error {
+				actual := &corev1.Secret{}
+				if err := client.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: "ns", Name: "pull-secret"}, actual); err != nil {
+					return err
+				}
+				expected := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ns",
+						Name:      "pull-secret",
+						Labels:    map[string]string{api.DPTPRequesterLabel: "bar"},
+					},
+					Data:      map[string][]byte{corev1.DockerConfigJsonKey: []byte("abc")},
+					Type:      corev1.SecretTypeDockerConfigJson,
+					Immutable: &im,
+				}
+				if diff := cmp.Diff(expected, actual, testhelper.RuntimeObjectIgnoreRvTypeMeta); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name:   "the target secret is update",
+			client: fakeclient.NewFakeClient(srcSecret.DeepCopy()),
+			verify: func(client ctrlruntimeclient.Client) error {
+				actual := &corev1.Secret{}
+				if err := client.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: "ns", Name: "pull-secret"}, actual); err != nil {
+					return err
+				}
+				expected := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ns",
+						Name:      "pull-secret",
+						Labels:    map[string]string{api.DPTPRequesterLabel: "bar"},
+					},
+					Data:      map[string][]byte{corev1.DockerConfigJsonKey: []byte("abc")},
+					Type:      corev1.SecretTypeDockerConfigJson,
+					Immutable: &im,
+				}
+				if diff := cmp.Diff(expected, actual, testhelper.RuntimeObjectIgnoreRvTypeMeta); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name:   "labels are ignored",
```

Comment:
Interesting.
https://github.com/openshift/ci-tools/blob/b3a4dbe39adc4d0461054019701cb41d9d6788ef/pkg/util/secret.go#L60

I cannot justify with the upstream's source: we wont enforce the labels.